kubernetes-sigs/aws-fsx-openzfs-csi-driver
# Add support for node startup taints

**Is this a bug fix or adding new feature?**
Feature

**What is this PR about? / Why do we need it?**
This PR allows users to configure the `fsx.openzfs.csi.aws.com/agent-not-ready` taint on their nodes to prevent workload pods from using the CSI driver pods before they are ready. When the CSI driver is ready, it will patch the node to remove this taint, allowing workload pods to be scheduled on the node.

**What testing is done?** 
Automated testing:
`make test` and `make test-sanity`

Manual testing:
1. Applied taints to the nodes in a cluster via 
`kubectl taint nodes $NODE_NAME fsx.openzfs.csi.aws.com/agent-not-ready:NoExecute`. Verified that the CSI driver removed this taint on startup. 
2. Applied unrelated taints to nodes in a cluster. Verified that these taints were not removed from the nodes on CSI driver start up.
3. Applied no taints to a cluster. 

In each case, I tested dynamic provisioning and mounting to verify this functionality was kept intact.
